torch/csrc/DynamicTypes.cpp

```diff
@@ -116,15 +94,33 @@ bool isStorage(PyObject* obj) {
   if (PyObject_TypeCheck(obj, getTypedStorageTypeObject())) {
     return true;
   }
-  auto obj_type = Py_TYPE(obj);
+  // TODO: Find look at the call sites of isStorage to find out if this change
+  // could break any of them
+  return THPStorage_Check(obj);
+}
+
+at::ScalarType getStorageType(PyObject* obj, bool& is_typed_storage) {
+  at::ScalarType scalar_type{at::ScalarType::Undefined};
+  is_typed_storage = PyObject_TypeCheck(obj, getTypedStorageTypeObject());
+
+  if (is_typed_storage) {
+    PyObject* dtype_obj = PyObject_GetAttrString(obj, "dtype");
+    TORCH_INTERNAL_ASSERT(dtype_obj);
+    Py_DECREF(dtype_obj);
```

Comment:
This decref seems too early. In practice it will be OK because obj probably holds a strong reference, but it might not (e.g., custom `__getattr__` returns temporary)